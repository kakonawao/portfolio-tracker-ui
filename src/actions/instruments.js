import fetch from "cross-fetch";

import {getRequestOptions, handleResponse} from "./util";
import { getEndpoint, PATHS } from "../config";


export const REQUEST_INSTRUMENTS = "REQUEST_INSTRUMENTS";
export const RECEIVE_INSTRUMENTS = "RECEIVE_INSTRUMENTS";
export const FAIL_INSTRUMENTS = "FAIL_INSTRUMENTS";

export const requestInstruments = () => {
    return {
        type: REQUEST_INSTRUMENTS
    }
};

export const failInstruments = (error) => {
    return {
        type: FAIL_INSTRUMENTS,
        error
    }
};

export const receiveInstruments = (data) => {
    return {
        type: RECEIVE_INSTRUMENTS,
        data
    }
};

export const fetchInstruments = (session) => {
    return function (dispatch) {

        dispatch(requestInstruments());

        return fetch(getEndpoint(PATHS.INSTRUMENTS), getRequestOptions(session))
            .then(response => handleResponse(response))
            .then(data => dispatch(receiveInstruments(data)))
            .catch(error => dispatch(failInstruments(error)));
    }
};
