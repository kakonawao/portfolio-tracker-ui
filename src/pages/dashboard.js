import React from "react";
import { connect } from "react-redux";

import AccountsSection from "../components/accountsSection";
import TransactionsSection from "../components/transactionsSection";
import SessionManager from "../components/sessionManager";


const mapStateToProps = (state) => {
    return {
        session: state.session
    }
};

class DashboardPage extends React.Component {

    render() {
        if (this.props.session.authenticated) {
            return (
                <div>
                    <AccountsSection/>
                    <TransactionsSection/>
                </div>
            );
        }

        return (
            <div>
                <SessionManager />
            </div>
        );
    }
}

export default connect(mapStateToProps)(DashboardPage);
