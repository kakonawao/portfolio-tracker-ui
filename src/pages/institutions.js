import React from "react";


export class InstitutionsPage extends React.Component {

    render() {
        return (
            <h2>Institutions</h2>
        )
    }
}

export default InstitutionsPage;
