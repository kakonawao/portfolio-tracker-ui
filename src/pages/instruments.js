import React from "react";


export class InstrumentsPage extends React.Component {

    render() {
        return (
            <h2>Instruments</h2>
        )
    }
}

export default InstrumentsPage;
