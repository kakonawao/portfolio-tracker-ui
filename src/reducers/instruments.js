import {
    FAIL_INSTRUMENTS,
    RECEIVE_INSTRUMENTS,
    REQUEST_INSTRUMENTS
} from "../actions/instruments";


const initialState = {
    loading: false,
    error: null,
    items: []
};

export const instrumentsReducer = (state = initialState, action) => {
    switch(action.type) {
        case REQUEST_INSTRUMENTS:
            return {
                ...state,
                error: null,
                loading: true
            };
        case FAIL_INSTRUMENTS:
            return {
                ...state,
                error: action.error,
                loading: false
            };
        case RECEIVE_INSTRUMENTS:
            return {
                ...state,
                loading: false,
                items: action.data
            };
        default:
            return state;
    }
};
