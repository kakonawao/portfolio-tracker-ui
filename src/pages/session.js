import React from "react";
import { connect } from "react-redux";

import LoginForm from "../components/loginForm";
import { login } from "../actions/session";


const mapStateToProps = (state) => {
    return {
        authenticated: state.session.authenticated,
        username: state.session.username
    }
};

export class SessionPage extends React.Component {

    constructor(props) {
        super(props);

        this.inputs = {
            username: React.createRef(),
            password: React.createRef()
        }
    }

    login = (event) => {
        event.preventDefault();
        const loginData = {
            username: this.inputs.username.current.value,
            password: this.inputs.password.current.value
        };
        this.props.dispatch(login(loginData));
    };

    renderUserData() {
        return (
            <div>
                <h2>Session</h2>
                <p>You are logged in as {this.props.username}.</p>
            </div>
        )
    }

    renderLoginForm() {
        return (
            <div>
                <h2>Session</h2>
                <LoginForm login={this.login}
                           username={this.inputs.username}
                           password={this.inputs.password}/>
            </div>
        );
    }

    render() {
        if (this.props.authenticated) {
            return this.renderUserData();
        }

        return this.renderLoginForm();
    };
}

export default connect(mapStateToProps)(SessionPage);
