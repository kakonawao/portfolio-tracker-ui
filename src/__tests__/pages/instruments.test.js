import React from "react";
import { shallow } from "enzyme";

import { InstrumentsPage } from "../../pages/instruments";


describe("<InstrumentsPage>", () => {

    it("renders", () => {
        const props = {
            dispatch: jest.fn()
        };

        const wrapper = shallow(<InstrumentsPage {...props}/>);
        expect(wrapper).toMatchSnapshot();
    })
});
