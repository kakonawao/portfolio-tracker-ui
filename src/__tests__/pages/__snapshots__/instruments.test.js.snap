// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InstrumentsPage> renders 1`] = `
<h2>
  Instruments
</h2>
`;
